pkg/api/v1/atlasdeployment_types.go

```diff
@@ -70,6 +71,14 @@ type AtlasDeploymentSpec struct {
 	ProcessArgs *ProcessArgs `json:"processArgs,omitempty"`
 }
 
+type SearchNode struct {
+	// +kubebuilder:validation:Enum:=S20_HIGHCPU_NVME;S30_HIGHCPU_NVME;S40_HIGHCPU_NVME;S50_HIGHCPU_NVME;S60_HIGHCPU_NVME;S70_HIGHCPU_NVME;S80_HIGHCPU_NVME;S30_LOWCPU_NVME;S40_LOWCPU_NVME;S50_LOWCPU_NVME;S60_LOWCPU_NVME;S80_LOWCPU_NVME;S90_LOWCPU_NVME;S100_LOWCPU_NVME;S110_LOWCPU_NVME
+	InstanceSize string `json:"instanceSize,omitempty"`
+	// +kubebuilder:validation:Minimum:=2
+	// +kubebuilder:validation:Maximum:=32
+	NodeCount uint8 `json:"nodeCount,omitempty"`
```

Comment:
this does not have any documentation:
```
$ k explain atlasdeployments.atlas.mongodb.com.spec.deploymentSpec.searchNodes
GROUP:      atlas.mongodb.com
KIND:       AtlasDeployment
VERSION:    v1

FIELD: searchNodes <[]Object>

DESCRIPTION:
    Settings for Search Nodes for the cluster. Currently, at most one search
    node configuration may be defined.
    
FIELDS:
  instanceSize	<string>
    <no description>

  nodeCount	<integer>
    <no description>
```

Reply:
How does one make kubebuilder pass the annotations from Go comments into the explain output?

Reply:
@josvazg this is done in API-Server by introspecting the CRD definitions, hence it is a side-effect of generating CRDs.

Reply:
I meant, what do we have to put in the Go code so that the generated CRDs will include documentation when using explain. I assume there are some special annotations to use or something.

Reply:
yes, see https://github.com/mongodb/mongodb-atlas-kubernetes/pull/1392/commits/728e0fb5ca88b0f2bfde945572bc452363df46f2 (part of this PR)